packages/line/src/LineSlices.js

```diff
@@ -11,30 +11,62 @@ import PropTypes from 'prop-types'
 import pure from 'recompose/pure'
 import LineSlicesItem from './LineSlicesItem'
 
-const LineSlices = ({
-    slices,
-    height,
-    showTooltip,
-    hideTooltip,
-    theme,
-    tooltip,
-    tooltipFormat,
-}) => (
-    <g>
-        {slices.map(slice => (
-            <LineSlicesItem
-                key={slice.id}
-                slice={slice}
-                height={height}
-                showTooltip={showTooltip}
-                hideTooltip={hideTooltip}
-                theme={theme}
-                tooltipFormat={tooltipFormat}
-                tooltip={tooltip}
-            />
-        ))}
-    </g>
-)
+class LineSlices extends React.Component {
+    render() {
+        const {
+            slices,
+            height,
+            showTooltip,
+            hideTooltip,
+            theme,
+            tooltip,
+            tooltipFormat,
+        } = this.props
+        // sliceWidth: half the distance from the previous point + half the distance to the next point
+        let sliceWidth
+        // xOffset: half the distance from the previous point
+        let xOffset
+        let lineSlicesItems = []
+        for (let i = 0; i < slices.length; i++) {
+            const currentXPosition = slices[i].data[0].position.x
+            const previousXPosition = i > 0 ? slices[i - 1].data[0].position.x : null
+            const nextXPosition = i < slices.length - 1 ? slices[i + 1].data[0].position.x : null
+            const widthBetweenCurrentAndNext = nextXPosition - currentXPosition
```

Comment:
What happens if two nulls are subtracted? It might be better to instead have zeroed values or have a minimum default for intended behavior?